python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -224,9 +224,14 @@ def find_parent_materialized_asset_partitions(
     target_asset_keys = target_asset_selection.resolve(asset_graph)
 
     for asset_key in target_asset_selection.upstream(depth=1).resolve(asset_graph):
-        records = instance_queryer.get_materialization_records(
-            asset_key=asset_key, after_cursor=latest_storage_id
-        )
+        # if the asset is non-partitioned, we only need the latest materialization
+        if not asset_graph.is_partitioned(asset_key):
+            latest_record = instance_queryer.get_latest_materialization_record(asset_key)
+            records = [latest_record] if latest_record else []
+        else:
+            records = instance_queryer.get_materialization_records(
```

Comment:
I think `get_event_records` has an `asset_partitions` argument to further filter these results. Hopefully this also helps with speeding up the query

Reply:
that's fair, and that definitely would help, but a future change will switch to using get_materialization_count_by_partition anyway, so I think I'll keep it this way in the short term